test/coding/test_embedded_ipython_code_executor.py

```diff
@@ -33,7 +34,7 @@ def test_create() -> None:
 @pytest.mark.skipif(skip, reason=skip_reason)
 def test_init() -> None:
     executor = EmbeddedIPythonCodeExecutor(timeout=10, kernel_name="python3", output_dir=".")
-    assert executor.timeout == 10 and executor.kernel_name == "python3" and executor.output_dir == "."
+    assert executor._timeout == 10 and executor._kernel_name == "python3" and executor._output_dir == Path(".")
```

Comment:
I added underscores for these private attributes. If this is not okay I can either revert or expose properties. I figured these are private so it makes sense.

Also, I changed _output_dir to be a Path

Reply:
Okay